### PR TITLE
kcov 42

### PR DESCRIPTION
**Pull request by p-linnane**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by p-linnane** [comment]
Opened an upstream issue for the Big Sur build failures: https://github.com/SimonKagstrom/kcov/issues/403

**Comment by SimonKagstrom** [comment]
When the build is fixed (I believe the last commit might do that), there is also the need for codesigning in v42 onwards.

The build does an ad hoc codesign, but I'm not sure if that works on other machines than the build one. If it doesn't, codesign needs to be run during the install stage, as per the discussion here: https://github.com/orgs/Homebrew/discussions/4725#discussioncomment-6755050

Basically something like,

```
codesign -s - --entitlements /usr/local/share/kcov/osx-entitlements.xml -f /usr/local/bin/kcov
```

assuming osx-entitlements.xml from the kcov source dir is installed in /usr/local/share/kcov by the homebrew installer.

**Comment by p-linnane** [comment]
Thank you for jumping in here @SimonKagstrom. Confirming that the Big Sur build is now fixed. We can take a look at the code signature piece once these bottles are published.

**Review by chenrui333 — APPROVED** [review]

**Comment by p-linnane** [comment]
@SimonKagstrom I'm seeing these bottles have the ad-hoc signature and one allowed entitlement: `com.apple.security.cs.debugger`

**Comment by SimonKagstrom** [comment]
Great, thanks for the very quick homebrew version bump!

Are you able to test it with a compiled program (e.g., a hello world in C, compiled with -g)? The debugger stuff is only needed with those.

**Comment by SimonKagstrom** [comment]
@p-linnane I tried v42 from the homebrew repo, and indeed it needs to be locally signed:

```
Simons-iMac:kcov ska$ /usr/local/bin/kcov /tmp/kcov ~/projects/build/calc/calc 15
kcov: error: task_for_pid failed with 5

[...]
Simons-iMac:kcov ska$ codesign -s - --entitlements ~/projects/kcov/osx-entitlements.xml -f /usr/local/bin/kcov
/usr/local/bin/kcov: replacing existing signature
Simons-iMac:kcov ska$ /usr/local/bin/kcov /tmp/kcov ~/projects/build/calc/calc 15+45-3
57
```
